include/librealsense2/hpp/rs_options.hpp

```diff
@@ -8,6 +8,25 @@
 
 namespace rs2
 {
+    template< class T >
+    class option_value_update_callback : public rs2_option_value_update_callback
+    {
+        T _callback;
+
+    public:
+        explicit option_value_update_callback( T callback )
+            : _callback( callback )
+        {
+        }
+
+        void on_value_changed( const rs2_options_list * list ) override
+        {
+            _callback( list );
+        }
+
+        void release() override { delete this; }
```

Comment:
Can we delete `this` from a class function?

Reply:
Yes, we do it in a few places just like this.